src/redux/books/books.js

```diff
@@ -15,12 +18,38 @@ export const removeBook = (id) => ({
   id,
 });
 
+export const getBook = (payload) => ({
+  type: GET_BOOKS,
+  payload,
+});
+
+export const addBookApi = (book) => async (dispatch) => {
+  await postBook(book);
+  dispatch(addBook(book));
+};
+
+export const removeBookApi = (id) => async (dispatch) => {
+  await deleteBook(id);
+  dispatch(removeBook(id));
+};
+
+export const getBookApi = () => async (dispatch) => {
+  const books = await getBooks();
+  const booklist = [];
+  Object.keys(books).forEach((id) => {
+    booklist.push({ item_id: id, title: books[id][0].title, category: books[id][0].category });
+  });
+  dispatch(getBook(booklist));
+};
+
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
       return [...state, action.payload];
     case REMOVE_BOOK:
```

Comment:
- [ ] When I click on the remove button all the cards are gone it's supposed to just one should be removed.

Reply:
Hi, @eri8-9.  I was adding items after sending the PR, that's the only reason I could imagine that were added items that weren't posted. I'll check the other comments.